tools/ansible/tasks/wsu/main.yaml

```diff
@@ -2,29 +2,131 @@
 - hosts: localhost
   vars:
     project_root: "{{ playbook_dir }}/../../../.."
-    kubelet_location: "https://dl.k8s.io/v1.16.2/kubernetes-node-windows-amd64.tar.gz"
     wmcb_exe: "wmcb.exe"
+    hybrid_overlay_exe : "hybrid-overlay.exe"
+    build: False
+    # Dictionary mapping Kubernetes version to the corresponding OpenShift Version
+    map_kubernetes_to_openshift_version:
+      '1.16': 4.3
+      '1.17': 4.4
+      '1.18': 4.5
 
   tasks:
-    - name: Gather required files
-      block:
+    - name: Create directory to store files to be used later
+      tempfile:
+        state: directory
+      register: tmp_dir
 
-        - name: Create directory to store files to be used later
-          tempfile:
-            state: directory
-          register: tmp_dir
+    - name: Get kubernetes version
+      shell: "oc version -o json | jq -r '.serverVersion.gitVersion'"
+      register: kubernetes_version
 
-        - name: Build WMCB
-          make:
-            target: build
-            chdir: "{{ project_root }}"
+    # Example of kubelet download url: "https://dl.k8s.io/v1.16.2/kubernetes-node-windows-amd64.tar.gz"
+    - name: Get kubelet download url
+      shell: "echo https://dl.k8s.io/{{ kubernetes_version.stdout }}/kubernetes-node-windows-amd64.tar.gz"
+      register: kubelet_location
+
+    # kubernetes_version v1.17.2 should correspond to 1.17 for the dictionary lookup
+    - name: Get kubernetes version without patch version
+      shell: "version={{ kubernetes_version.stdout }} ; echo ${version%.*} | tr -d v"
+      register: kubernetes_version_without_patch
+
+    # Changelog url to lookup SHA for kube node binaries
+    # Example: https://github.com/kubernetes/kubernetes/blob/release-1.17/CHANGELOG/CHANGELOG-1.17.md
+    - name: Get kubernetes changelog url
+      shell: |
+        echo "https://github.com/kubernetes/kubernetes/blob/release-"\
+        "{{ kubernetes_version_without_patch.stdout }}/CHANGELOG/CHANGELOG-"\
+        "{{ kubernetes_version_without_patch.stdout }}.md"
+      register: changelog
+
+    - name: Get kube node binary SHA
+      shell: |
+        curl -s "{{ changelog.stdout }}" | \
+        grep -A 1 "{{ kubernetes_version.stdout }}" | \
+        grep -A 1 node-windows | tail -n 1 | sed -e 's/.*<code>\(.*\)<\/code>.*/\1/'
+      register: kube_node_sha
+
+    # OpenShift cluster version
+    - name: Register cluster version
+      shell: "echo {{ map_kubernetes_to_openshift_version[kubernetes_version_without_patch.stdout] }}"
```

Comment:
What happens if the version is not present in the map?

Reply:
Added a guard for this